WalletWasabi.Tests/UnitTests/TestWallet.cs

```diff
@@ -0,0 +1,108 @@
+using NBitcoin;
+using NBitcoin.Crypto;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.BitcoinCore.Rpc;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Crypto;
+using WalletWasabi.WabiSabi.Client;
+
+namespace WalletWasabi.Tests.UnitTests;
+
+public class TestWallet : IKeyChain, IDestinationProvider
+{
+	public TestWallet(string name, IRPCClient rpc)
+	{
+		Rpc = rpc;
+		Key = new Key(Hashes.SHA256(Encoding.ASCII.GetBytes(name)));
```

Comment:
Maybe it's better to use UTF-8:

```suggestion
		Key = new Key(Hashes.SHA256(Encoding.UTF8.GetBytes(name)));
```

Reply:
Also `Key` is disposable so the class should be disposable imo.

Reply:
You're right. I have to make this class disposable, f***, Nicolas ruined my life.